src/js/jsx/search/SearchBar.jsx

```diff
@@ -500,20 +500,29 @@ define(function (require, exports, module) {
 
         render: function () {
             var searchStrings = nls.localize("strings.SEARCH"),
-                noMatchesOption = {
+                filters = this.state.filter,
+                placeholderText,
+                clearInputBtn;
+
+            var flux = this.getFlux(),
+                searchStore = flux.store("search"),
+                registeredSearchModules = searchStore._registeredSearchTypes;
```

Comment:
Even though JS allows us to get any variables of stores, it's a bad convention to directly access them. Please implement a getter method in search store that will return you the correct module payload for a given filter and use that to get your `searchPayload`